src/Lachain.Storage/Repositories/SnapshotIndexRepository.cs

```diff
@@ -87,5 +96,113 @@ private void SetVersion(uint repository, ulong block, ulong version)
                 version.ToBytes()
             );
         }
+
+        private void DeleteVersion(uint repository, ulong block, ulong version)
+        {
+            //Console.WriteLine($"Deleting version {version} for "
+            //    + $"{(RepositoryType) repository} for block {block}");
+            _dbContext.Delete(
+                EntryPrefix.SnapshotIndex.BuildPrefix(
+                    repository.ToBytes().Concat(block.ToBytes()).ToArray()));
+        }
+
+        private void SetDeleteOldSnapshotStatus(byte status)
+        {
+            deleteOldSnapshotStatus = status;
+            var prefix = EntryPrefix.DeleteOldSnapshotStatus.BuildPrefix();
+            var content = new byte[1];
+            content[0] = status;
+            _dbContext.Save(prefix, content);
+        }
+
+        public void DeleteOldSnapshot(ulong depth, ulong totalBlocks)
+        {
+            if (DeleteOldSnapshotStatus <= 1)
+            {
+                // set deletingOldSnapshot = 1 when the process starts
+                SetDeleteOldSnapshotStatus(1);
+                Console.WriteLine($"Keeping latest {depth+1} snapshots from last approved snapshot" 
+                    + $"for blocks: {totalBlocks - depth} to {totalBlocks}");
+
+                // saving nodes for recent (depth + 1) snapshots temporarily
+                // so that all remaining nodes can be deleted
+                UpdateNodeIdToBatch(totalBlocks, depth, true);
+            }
+
+            if (DeleteOldSnapshotStatus <= 2)
+            {
+                // set deletingOldSnapshot = 2 when deleting old nodes
+                SetDeleteOldSnapshotStatus(2);
+                // deleting all nodes that are not reachable from recent (depth+1) snapshots
+                ulong deletedNodes = 0;
+                Console.WriteLine("Deleting nodes from DB that are not reachable from recent snapshots");
```

Comment:
We should write it in the log instead of console. What do you think ?

Reply:
Kept log records.